packages/wp-checkout/src/components/checkout-step.js

```diff
@@ -0,0 +1,90 @@
+/* @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import joinClasses from '../lib/join-classes';
+import { useLocalize } from '../lib/localize';
+
+export default function CheckoutStep( {
+	className,
+	stepNumber,
+	title,
+	onEdit,
+	collapsed,
+	children,
+} ) {
+	const classNames = collapsed
+		? [ className, 'checkout-step', 'checkout-step--is-collapsed' ]
+		: [ className, 'checkout-step' ];
+	return (
+		<div className={ joinClasses( classNames ) }>
+			<CheckoutStepHeader
+				stepNumber={ stepNumber }
+				title={ title }
+				collapsed={ collapsed }
+				onEdit={ onEdit }
+			/>
+			{ children }
+		</div>
+	);
+}
+
+CheckoutStep.propTypes = {
+	className: PropTypes.string,
+	stepNumber: PropTypes.number.isRequired,
```

Comment:
Should we validate which step numbers make sense? E.g. what if `stepNumber` is -1, or 1/2, or NaN?
If this is never exposed to the host page maybe it doesn't matter.

Reply:
> Should we validate which step numbers make sense? E.g. what if stepNumber is -1, or 1/2, or NaN?

That's a good idea! But I think I'd rather put that validation into the actual code, since propTypes only show a warning, and even then only in development. We can throw an Error if the prop is invalid.

> If this is never exposed to the host page maybe it doesn't matter.

It probably won't be when using the `Checkout` component, but `CheckoutStep` is going to be part of the public API so we should make it as helpful as possible.